helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java

```diff
@@ -187,6 +189,13 @@ public Response createCluster(@PathParam("clusterId") String clusterId,
       LOG.error("Failed to create cluster {}. Exception: {}.", clusterId, ex);
       return serverError(ex);
     }
+
+    try {
```

Comment:
If we perform this before the cluster creation, then if ACL fails the cluster isn't created either. I'm not sure which one is the better option though, a cluster without ACL or an ACL without a cluster.

Reply:
Synced offline: we will allow only allow a cluster to be created if the ACL exists, so we are changing the order. 

When creating ACL, we will keep it idempotent, but only for the correct user.